### PR TITLE
codec(ticdc): simple protocol encode index for the enum and set type

### LINKED ISSUE #10661: simple protocol encode enum and set index value
For the enum and set, encode the index and mask integer value. Since the enum and set definition can be found in the DDL message, the consumer can use the index to find the concrete value

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What problem does this PR solve?

Issue Number: close #10661

### What is changed and how it works?

* For the enum and set type, encode the index and mask integer into the dml message

#### Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
`None`.
```

### CODE REVIEW

**Review by asddongmen — APPROVED** [review]

File: pkg/sink/codec/common/helper.go [hunk 1]
```diff
@@ -132,27 +132,27 @@ func SnapshotQuery(
...
 // by follow https://github.com/pingcap/tidb/blob/e3417913f58cdd5a136259b902bf177eaf3aa637/types/binary_literal.go#L105
-func BinaryLiteralToInt(bytes []byte) (uint64, error) {
+func MustBinaryLiteralToInt(bytes []byte) uint64 {
 	bytes = trimLeadingZeroBytes(bytes)
 	length := len(bytes)
 
 	if length > 8 {
-		log.Error("invalid bit value found", zap.ByteString("value", bytes))
-		return math.MaxUint64, errors.New("invalid bit value")
+		log.Panic("invalid bit value found", zap.ByteString("value", bytes))
```
[the remark this answers is not in the document]
Comment (3AceShowHand) [reply pkg/sink/codec/common/helper.go:142]: The bytes comes from the mounter, it won't cause any error I think, even if the error, we can do nothing.

**Review by CharlesCheung96 — APPROVED** [review]

File: pkg/sink/codec/common/helper.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (asddongmen) [reply pkg/sink/codec/common/helper.go:142]: Error only affect one changefeed, panic crash the whole server.

**Pull request merged**